tests/catch/unit/errorHandling/hipGetLastError.cc

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <hip_test_common.hh>
+#include <hip/hip_runtime_api.h>
+#include <threaded_zig_zag_test.hh>
+
+TEST_CASE("Unit_hipGetLastError_Positive_Basic") {
+  HIP_CHECK(hipGetLastError());
+  HIP_CHECK_ERROR(hipMalloc(nullptr, 1), hipErrorInvalidValue);
+  HIP_CHECK_ERROR(hipGetLastError(), hipErrorInvalidValue);
+  HIP_CHECK(hipGetLastError());
+}
+
+TEST_CASE("Unit_hipGetLastError_Positive_Threaded") {
+  class HipGetLastErrorThreadedTest : public ThreadedZigZagTest<HipGetLastErrorThreadedTest> {
```

Comment:
Where's ThreadedZigZagTest defined? I can't find the file threaded_zig_zag_test.hh

Reply:
threaded_zig_zag_test.hh has its own PR (#2887) to avoid later merge conflicts. This means that the PRs referencing threaded_zig_zag_test.hh depend on #2887 being merged first.

Reply:
OK, makes sense, thanks!